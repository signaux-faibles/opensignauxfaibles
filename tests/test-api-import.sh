#!/bin/bash

# Test de bout en bout de l'API "import".
# Référence: https://github.com/signaux-faibles/documentation/blob/master/processus-traitement-donnees.md#vue-densemble-des-canaux-de-transformation-des-donn%C3%A9es
# Ce script doit être exécuté depuis la racine du projet. Ex: par test-all.sh.

tests/helpers/mongodb-container.sh stop

set -e # will stop the script if any command fails with a non-zero exit code

# Setup
FLAGS="$*" # the script will update the golden file if "--update" flag was provided as 1st argument
TMP_DIR="tests/tmp-test-execution-files"
OUTPUT_FILE="${TMP_DIR}/test-api-import.output.txt"
GOLDEN_FILE="tests/output-snapshots/test-api-import.golden.txt"
mkdir -p "${TMP_DIR}"

# Clean up on exit
function teardown {
    tests/helpers/dbmongo-server.sh stop || true # keep tearing down, even if "No matching processes belonging to you were found"
    tests/helpers/mongodb-container.sh stop
}
trap teardown EXIT

PORT="27016" tests/helpers/mongodb-container.sh start

MONGODB_PORT="27016" tests/helpers/dbmongo-server.sh setup

echo ""
echo "📝 Inserting test data..."
sleep 1 # give some time for MongoDB to start

tests/helpers/mongodb-container.sh run > /dev/null << CONTENTS
  db.Admin.insertOne({
    "_id" : {
        "key" : "1910",
        "type" : "batch"
    },
    "files": {
      "admin_urssaf": [
        "/../lib/urssaf/testData/comptesTestData.csv"
      ],
      "delai": [
        "/../lib/urssaf/testData/delaiTestData.csv"
      ]
    },
    "param" : {
        "date_debut" : ISODate("2001-01-01T00:00:00.000+0000"),
        "date_fin" : ISODate("2019-02-01T00:00:00.000+0000")
    }
  })
CONTENTS

echo ""
echo "💎 Parsing and importing data thru dbmongo API..."
tests/helpers/dbmongo-server.sh start
echo "- POST /api/data/import 👉 $(http --print=b --ignore-stdin :5000/api/data/import batch=1910 parsers:='["delai"]')"

OUTPUT_GZ_FILE=dbmongo/$(http --print=b --ignore-stdin :5000/api/data/validate collection=ImportedData | tr -d '"')
echo "- POST /api/data/validate 👉 ${OUTPUT_GZ_FILE}"

(tests/helpers/mongodb-container.sh run \
  | perl -p -e 's/"[0-9a-z]{32}"/"______________Hash______________"/' \
  | perl -p -e 's/"[0-9a-z]{24}"/"________ObjectId________"/' \
  > "${OUTPUT_FILE}" \
) << CONTENT
print("// Documents from db.ImportedData, after call to /api/data/import:");
printjson(db.ImportedData.find().sort({"value.key":1}).toArray());
print("// Documents from db.Journal:");
printjson(db.Journal.find({ "event.report": { "\$exists": true } }).toArray().map(doc => ({
  // note: we use map() to force the order of properties at every run of this test
  event: {
    headFilters: doc.event.headFilters,
    headErrors: doc.event.headErrors,
    headFatal: doc.event.headFatal,
    report: doc.event.report,
    batchKey: doc.event.batchKey
  },
  code: doc.code
})));

print("// Results of call to /api/data/validate:");
CONTENT

zcat < "${OUTPUT_GZ_FILE}" \
  | perl -p -e 's/"[0-9a-z]{32}"/"______________Hash______________"/' \
  | perl -p -e 's/"[0-9a-z]{24}"/"________ObjectId________"/' \
  | sort \
  >> "${OUTPUT_FILE}"

# Display JS errors logged by MongoDB, if any
tests/helpers/mongodb-container.sh exceptions || true

tests/helpers/diff-or-update-golden-master.sh "${FLAGS}" "${GOLDEN_FILE}" "${OUTPUT_FILE}"

rm "${OUTPUT_GZ_FILE}"
rm -rf "${TMP_DIR}"
# Now, the "trap" commands will clean up the rest.
