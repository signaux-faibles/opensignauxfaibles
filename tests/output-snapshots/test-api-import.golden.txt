// Documents from db.ImportedData, after call to /api/data/import:
[
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "",
			"batch" : {
				"1910" : {
					"compte" : {
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "",
							"numero_compte" : "450359886246036238",
							"periode" : ISODate("_______ Date _______")
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "00000000000000",
			"batch" : {
				"1910" : {
					"compte" : {
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "00000000000000",
							"numero_compte" : "111982477292496174",
							"periode" : ISODate("_______ Date _______")
						}
					},
					"delai" : {
						"______________Hash______________" : {
							"numero_compte" : "111982477292496174",
							"numero_contentieux" : "5856690802766",
							"date_creation" : ISODate("1999-12-31T23:00:00Z"),
							"date_echeance" : ISODate("2001-01-03T23:00:00Z"),
							"duree_delai" : 456,
							"denomination" : "etmlgcthph ecoowixknaae se",
							"indic_6m" : "lhm",
							"annee_creation" : 5647,
							"montant_echeancier" : 5065.79,
							"stade" : "nbz sk",
							"action" : "cya ng"
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "entreprise",
			"key" : "000111222",
			"batch" : {
				"1910" : {
					"bdf" : {
						"______________Hash______________" : {
							"siren" : "000111222",
							"annee_bdf" : 2013,
							"arrete_bilan_bdf" : ISODate("2012-12-31T00:00:00Z"),
							"raison_sociale" : "SOCIETE",
							"secteur" : "01 Industrie et Transports (Hors TRM)",
							"poids_frng" : -37.5,
							"taux_marge" : -63.1,
							"delai_fournisseur" : 88.5,
							"dette_fiscale" : 561,
							"financier_court_terme" : 5.2,
							"frais_financier" : -9.4
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "entreprise",
			"key" : "000111223",
			"batch" : {
				"1910" : {
					"bdf" : {
						"______________Hash______________" : {
							"siren" : "000111223",
							"annee_bdf" : 2014,
							"arrete_bilan_bdf" : ISODate("2013-12-31T00:00:00Z"),
							"raison_sociale" : "SOCIETE",
							"secteur" : "01 Industrie et Transports (Hors TRM)",
							"poids_frng" : -77.1,
							"taux_marge" : -13.9,
							"delai_fournisseur" : 203.8,
							"dette_fiscale" : 204,
							"financier_court_terme" : 9.2,
							"frais_financier" : 7.3
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "entreprise",
			"key" : "000111224",
			"batch" : {
				"1910" : {
					"bdf" : {
						"______________Hash______________" : {
							"siren" : "000111224",
							"annee_bdf" : 2015,
							"arrete_bilan_bdf" : ISODate("2014-12-31T00:00:00Z"),
							"raison_sociale" : "SOCIETE",
							"secteur" : "01 Industrie et Transports (Hors TRM)",
							"poids_frng" : 64.5,
							"taux_marge" : 103.5,
							"delai_fournisseur" : 22.9,
							"dette_fiscale" : 235,
							"financier_court_terme" : 7.3,
							"frais_financier" : 20.3
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "11111111111111",
			"batch" : {
				"1910" : {
					"compte" : {
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						},
						"______________Hash______________" : {
							"siret" : "11111111111111",
							"numero_compte" : "636043216536562844",
							"periode" : ISODate("_______ Date _______")
						}
					},
					"delai" : {
						"______________Hash______________" : {
							"numero_compte" : "636043216536562844",
							"numero_contentieux" : "8981265766679",
							"date_creation" : ISODate("2000-03-02T23:00:00Z"),
							"date_echeance" : ISODate("2001-05-04T22:00:00Z"),
							"duree_delai" : 758,
							"denomination" : "mtj n.p. eg saxjoc",
							"indic_6m" : "uqu",
							"annee_creation" : 9941,
							"montant_echeancier" : 74825.41,
							"stade" : "qh",
							"action" : "bad mu"
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "15756082097503",
			"batch" : {
				"1910" : {
					"apdemande" : {
						"______________Hash______________" : {
							"id_demande" : "f288626887",
							"effectif_entreprise" : 559,
							"effectif" : 0,
							"date_statut" : ISODate("2012-02-11T00:00:00Z"),
							"periode" : {
								"start" : ISODate("2017-08-09T00:00:00Z"),
								"end" : ISODate("2017-09-09T00:00:00Z")
							},
							"hta" : 42078.52,
							"mta" : 666113.56,
							"effectif_autorise" : 99,
							"motif_recours_se" : 8,
							"heure_consommee" : 48512,
							"montant_consommee" : 689663.3,
							"effectif_consomme" : 319
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "50930194570891",
			"batch" : {
				"1910" : {
					"apdemande" : {
						"______________Hash______________" : {
							"id_demande" : "j354624024",
							"effectif_entreprise" : 20,
							"effectif" : 82,
							"date_statut" : ISODate("2012-03-11T00:00:00Z"),
							"periode" : {
								"start" : ISODate("2005-08-29T00:00:00Z"),
								"end" : ISODate("2005-09-29T00:00:00Z")
							},
							"hta" : 178.53,
							"mta" : 953.96,
							"effectif_autorise" : 9,
							"motif_recours_se" : 3,
							"heure_consommee" : 81,
							"montant_consommee" : 820.94,
							"effectif_consomme" : 8
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "53312397597505",
			"batch" : {
				"1910" : {
					"apconso" : {
						"______________Hash______________" : {
							"id_conso" : "p95a076375",
							"heure_consomme" : 6.25,
							"montant" : 27,
							"effectif" : 7,
							"periode" : ISODate("_______ Date _______")
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "61433760225578",
			"batch" : {
				"1910" : {
					"apconso" : {
						"______________Hash______________" : {
							"id_conso" : "d363954866",
							"heure_consomme" : 311.54,
							"montant" : 214,
							"effectif" : 0,
							"periode" : ISODate("_______ Date _______")
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "65272073959845",
			"batch" : {
				"1910" : {
					"apconso" : {
						"______________Hash______________" : {
							"id_conso" : "u90q913412",
							"heure_consomme" : 44.11,
							"montant" : 659,
							"effectif" : 2,
							"periode" : ISODate("_______ Date _______")
						}
					}
				}
			}
		}
	},
	{
		"_id" : ObjectId("________ObjectId________"),
		"value" : {
			"scope" : "etablissement",
			"key" : "99508628226971",
			"batch" : {
				"1910" : {
					"apdemande" : {
						"______________Hash______________" : {
							"id_demande" : "m80v947695",
							"effectif_entreprise" : 949,
							"effectif" : 535,
							"date_statut" : ISODate("2012-04-11T00:00:00Z"),
							"periode" : {
								"start" : ISODate("2014-04-29T00:00:00Z"),
								"end" : ISODate("2014-05-29T00:00:00Z")
							},
							"hta" : 8921.61,
							"mta" : 72697.76,
							"effectif_autorise" : 83,
							"motif_recours_se" : 7,
							"heure_consommee" : 6165,
							"montant_consommee" : 1456.27,
							"effectif_consomme" : 64
						}
					}
				}
			}
		}
	}
]
// Documents from db.Journal:
[
	{
		"event" : {
			"headFilters" : [ ],
			"headErrors" : [ ],
			"headFatal" : [ ],
			"report" : "/../lib/bdf/testData/bdfTestData.csv: intégration terminée, 3 lignes traitées, 0 erreures fatales, 0 rejets, 0 lignes filtrées, 3 lignes valides",
			"batchKey" : "1910"
		},
		"code" : "bdfParser"
	},
	{
		"event" : {
			"headFilters" : [ ],
			"headErrors" : [ ],
			"headFatal" : [ ],
			"report" : "Admin_urssaf: intégration terminée, 3 lignes traitées, 0 erreures fatales, 0 rejets, 0 lignes filtrées, 3 lignes valides",
			"batchKey" : "1910"
		},
		"code" : "compteParser"
	},
	{
		"event" : {
			"headFilters" : [
				"Cycle 2: Pas de siret associé au compte 450359886246036238 à la période 2019-01-23 00:00:00 +0000 UTC"
			],
			"headErrors" : [ ],
			"headFatal" : [ ],
			"report" : "/../lib/urssaf/testData/delaiTestData.csv: intégration terminée, 3 lignes traitées, 0 erreures fatales, 0 rejets, 1 lignes filtrées, 2 lignes valides",
			"batchKey" : "1910"
		},
		"code" : "delaiParser"
	},
	{
		"event" : {
			"headFilters" : [ ],
			"headErrors" : [ ],
			"headFatal" : [ ],
			"report" : "/../lib/apconso/testData/apconsoTestData.csv: intégration terminée, 3 lignes traitées, 0 erreures fatales, 0 rejets, 0 lignes filtrées, 3 lignes valides",
			"batchKey" : "1910"
		},
		"code" : "parserApconso"
	},
	{
		"event" : {
			"headFilters" : [ ],
			"headErrors" : [ ],
			"headFatal" : [ ],
			"report" : "/../lib/apdemande/testData/apdemandeTestData.csv: intégration terminée, 3 lignes traitées, 0 erreures fatales, 0 rejets, 0 lignes filtrées, 3 lignes valides",
			"batchKey" : "1910"
		},
		"code" : "parserApdemande"
	}
]
// Results of call to /api/data/validate:
{"_id":"________ObjectId________","batchKey":"1910","dataHash":"______________Hash______________","dataObject":{"action":"bad mu","annee_creation":9941,"date_creation":"2000-03-02T23:00:00Z","date_echeance":"2001-05-04T22:00:00Z","denomination":"mtj n.p. eg saxjoc","duree_delai":758,"indic_6m":"uqu","montant_echeancier":74825.41,"numero_compte":"636043216536562844","numero_contentieux":"8981265766679","stade":"qh"},"dataType":"delai"}
{"_id":"________ObjectId________","batchKey":"1910","dataHash":"______________Hash______________","dataObject":{"action":"cya ng","annee_creation":5647,"date_creation":"1999-12-31T23:00:00Z","date_echeance":"2001-01-03T23:00:00Z","denomination":"etmlgcthph ecoowixknaae se","duree_delai":456,"indic_6m":"lhm","montant_echeancier":5065.79,"numero_compte":"111982477292496174","numero_contentieux":"5856690802766","stade":"nbz sk"},"dataType":"delai"}
{"_id":"________ObjectId________","batchKey":"1910","dataHash":"______________Hash______________","dataObject":{"annee_bdf":2013,"arrete_bilan_bdf":"2012-12-31T00:00:00Z","delai_fournisseur":88.5,"dette_fiscale":561,"financier_court_terme":5.2,"frais_financier":-9.4,"poids_frng":-37.5,"raison_sociale":"SOCIETE","secteur":"01 Industrie et Transports (Hors TRM)","siren":"000111222","taux_marge":-63.1},"dataType":"bdf"}
{"_id":"________ObjectId________","batchKey":"1910","dataHash":"______________Hash______________","dataObject":{"annee_bdf":2014,"arrete_bilan_bdf":"2013-12-31T00:00:00Z","delai_fournisseur":203.8,"dette_fiscale":204,"financier_court_terme":9.2,"frais_financier":7.3,"poids_frng":-77.1,"raison_sociale":"SOCIETE","secteur":"01 Industrie et Transports (Hors TRM)","siren":"000111223","taux_marge":-13.9},"dataType":"bdf"}
{"_id":"________ObjectId________","batchKey":"1910","dataHash":"______________Hash______________","dataObject":{"annee_bdf":2015,"arrete_bilan_bdf":"2014-12-31T00:00:00Z","delai_fournisseur":22.9,"dette_fiscale":235,"financier_court_terme":7.3,"frais_financier":20.3,"poids_frng":64.5,"raison_sociale":"SOCIETE","secteur":"01 Industrie et Transports (Hors TRM)","siren":"000111224","taux_marge":103.5},"dataType":"bdf"}
